lib/scrabble.rb

```diff
@@ -1,4 +1,58 @@
 module Scrabble
-  # require your gems and classes here
-  # require_relative 'scrabble/whatevs'
+#   # require your gems and classes here
+#   # require_relative 'scrabble/whatevs'
+VALUES = {"a"=> 1, "b"=> 3, "c"=> 3, "d"=> 2, "e"=> 1, "f"=> 4, "g"=> 2, "h"=> 4, "i"=> 1, "j"=> 8, "k"=> 5, "l"=> 1, "m"=> 3, "n"=> 1, "o"=> 1, "p"=> 3, "q"=> 10, "r"=> 1, "s"=> 1, "t"=> 1, "u"=> 1, "v"=> 4, "w"=> 4, "x"=> 8, "y"=> 4, "z"=> 10}
+
+  class Scrabble
+
+    def self.score(word)
+      if word.length > 7 || word.length < 1
+        return "Error"
+      end
+
+      letters = %w(a b c d e f g h i j k l m n o p q r s t u v w x y z)
+      word_array = word.downcase.split(//) # makes an [] with each letter as its own element
+      word_array.each do |character|
+        if !letters.include?(character)
+          return "Error"
+        end
+      end
+
+      word_score = []
+      word.downcase.each_char do |letter|
+        VALUES.each do |key, value|
+          if letter == key
+            word_score.push(value)
+          end
+        end
+      end
+
+      score = word_score.reduce(0, :+)
+      return score
+    end
+
+    def self.highest_score_from(array_of_words)
+      array_of_values = []
+      array_of_words.collect do |word|
+        values = self.score(word)
+        array_of_values.push(values)
+      end
+      # return array_of_values
+      hash_of_words = Hash[array_of_words.zip array_of_values]#return hash_of_words with the words as the keys and their point value as the value
```

Comment:
Very interesting use of zip! You could also have created a hash in your loop as you retrieved each value like on line 38:
```
hash_of_words[word] = values
```